upstream: Porting core-protocol related upstream changes in range mainnet-1.43.1 to mainnet-1.44.3

muXxer opened the pull request:
# Description of change

This PR contains a batch of core-protocol related upstream changes. The changes have been individually reviewed before squash-merging. None of them modify protocol parameters, so a new protocol version is not necessary.

This PR should be rebased on top of develop without squashing as it already contains squashed PRs.

## Links to any relevant issues

List of included PRs:

- #8100
- #8096 
- #8099 
- #8115 
- #8132 
- #8136 
- #8144 
- #8135 
- #8153 
- #8155 
- #8163
- #8170
- #8143 
- #8282 
- #8247
- #8393 

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

- [x] Basic tests (linting, compilation, formatting, unit/integration tests)
- [ ] Patch-specific tests (correctness, functionality coverage)

bingyanglin committed:
upstream(core): Fix execution driver panic which became possible due to synchronous tx execution (#8100)

- Upstream range: [v1.43.1, v1.44.3)
- Port commit:
-
https://github.com/MystenLabs/sui/commit/b300c776acbc702fd888b7581f955bf1b8a56c02
- Description:

Fix execution driver panic which became possible due to synchronous tx
execution.
Crash found by antithesis.
Removed the retry loop which has been disabled everywhere except mainnet
for some time.

Part of #5727

- [x] Basic tests (linting, compilation, formatting, unit/integration
tests)
- [ ] Patch-specific tests (correctness, functionality coverage)
- [ ] I have added tests that prove my fix is effective or that my
feature works
- [ ] I have checked that new and existing unit tests pass locally with
my changes

jkrvivian committed:
upstream(core): Remove re-export of CheckpointData (#8096)

# Description of change

- Upstream range: [v1.43.1, v1.44.3)
- Port commit: 
-
https://github.com/MystenLabs/sui/commit/20df8b86c56b505febcc1ef2c2fd59a56476c292


## Links to any relevant issues

Part of #5727    

## How the change has been tested

- [x] Basic tests (linting, compilation, formatting, unit/integration
tests)
- [ ] Patch-specific tests (correctness, functionality coverage)
- [ ] I have added tests that prove my fix is effective or that my
feature works
- [ ] I have checked that new and existing unit tests pass locally with
my changes

bingyanglin committed:
upstream(core): Remove unsigned version of Discovery protocol (#8099)

# Description of change

- Upstream range: [v1.43.1, v1.44.3)
- Port commit: 
-
https://github.com/MystenLabs/sui/commit/314960e2e857e4011bc353aa56e017b0847202d2
- Description:

Remove unsigned version of Discovery protocol

## Links to any relevant issues

Part of #5727 

## How the change has been tested

- [x] Basic tests (linting, compilation, formatting, unit/integration
tests)
- [ ] Patch-specific tests (correctness, functionality coverage)
- [ ] I have added tests that prove my fix is effective or that my
feature works
- [ ] I have checked that new and existing unit tests pass locally with
my changes

### Release Notes

- [ ] Protocol:
- [x] Nodes (Validators and Full nodes): Remove the unsigned version of
the discovery protocol
- [ ] Indexer:
- [ ] JSON-RPC:
- [ ] GraphQL:
- [ ] CLI:
- [ ] Rust SDK:
- [ ] REST API:

jkrvivian committed:
upstream(core): Receive N previously-processed consensus commits at startup (#8115)

# Description of change

- Upstream range: [v1.43.1, v1.44.3)
- Port commit: 
-
https://github.com/MystenLabs/sui/commit/efd3e7e374947460cb423a219d832d8a041b15b8

NOTE: The upstream commit is based on changes of fastpath that we didn't
port. So the changes are a bit different from upstream.


## Links to any relevant issues
Close #6586 
Part of #5727 

## How the change has been tested

- [x] Basic tests (linting, compilation, formatting, unit/integration
tests)
- [ ] Patch-specific tests (correctness, functionality coverage)
- [ ] I have added tests that prove my fix is effective or that my
feature works
- [ ] I have checked that new and existing unit tests pass locally with
my changes

---------

Co-authored-by: muXxer <git@muxxer.de>

jkrvivian committed:
upstream(core): Remove indirect object functionality (#8132)

- Upstream range: [v1.43.1, v1.44.3)
- Port commit:
-
https://github.com/MystenLabs/sui/commit/5739b6d0e9359c40137f497a3509fca892d22811

The default value is `0`, so this functionality was never used. We can
safely port this commit to remove it.

Part of #5727

- [x] Basic tests (linting, compilation, formatting, unit/integration
tests)
- [ ] Patch-specific tests (correctness, functionality coverage)
- [ ] I have added tests that prove my fix is effective or that my
feature works
- [ ] I have checked that new and existing unit tests pass locally with
my changes

- [ ] Protocol:
- [x] Nodes (Validators and Full nodes):
   - Remove  `indirect-objects-threshold` from node config
- [ ] Indexer:
- [ ] JSON-RPC:
- [ ] GraphQL:
- [ ] CLI:
- [ ] Rust SDK:
- [ ] REST API:

bingyanglin committed:
upstream(core): Fix edge case in package safety check (#8136)

# Description of change

- Upstream range: [v1.43.1, v1.44.3)
- Port commit: 
-
https://github.com/MystenLabs/sui/commit/205ef11e0ac43c96938b7a214bcbc0b2016ac778
- Description:

Fix edge case in package safety check.
A new package may be in the dirty set but not flushed to disk

## Links to any relevant issues

Part of #5727 

## How the change has been tested

- [x] Basic tests (linting, compilation, formatting, unit/integration
tests)
- [ ] Patch-specific tests (correctness, functionality coverage)
- [ ] I have added tests that prove my fix is effective or that my
feature works
- [ ] I have checked that new and existing unit tests pass locally with
my changes

jkrvivian committed:
upstream(node): Remove keys reference in typed-store (#8144)

# Description of change

- Upstream range: [v1.43.1, v1.44.3)
- Port commit: 
-
https://github.com/MystenLabs/sui/commit/7a079dc8569d6beabf9bfbd08bdd62fa27b90e82

NOTE: We keep the `test_db.rs` to test the remaining functionalities.

## Links to any relevant issues
Part of #5727 

## How the change has been tested

- [x] Basic tests (linting, compilation, formatting, unit/integration
tests)
- [ ] Patch-specific tests (correctness, functionality coverage)
- [ ] I have added tests that prove my fix is effective or that my
feature works
- [ ] I have checked that new and existing unit tests pass locally with
my changes

jkrvivian committed:
upstream(core): Remove values and few other deprecated methods (#8135)

# Description of change

- Upstream range: [v1.43.1, v1.44.3)
- Port commit: 
-
https://github.com/MystenLabs/sui/commit/8d3abb7d78fed8888464ff196fabf96cbb063d03

The `delete_file_in_range` is not removed, because the upstream
accidentally removed it and later brought it back in
https://github.com/MystenLabs/sui/commit/5c8266fca307fc0ebb6463dfcd5a5bb541de4d41


## Links to any relevant issues
Part of #5727 

## How the change has been tested

- [x] Basic tests (linting, compilation, formatting, unit/integration
tests)
- [ ] Patch-specific tests (correctness, functionality coverage)
- [ ] I have added tests that prove my fix is effective or that my
feature works
- [ ] I have checked that new and existing unit tests pass locally with
my changes

bingyanglin committed:
upstream(core): Add notify_read_synced_checkpoint/notify_read_executed_checkpoint (#8153)

# Description of change

- Upstream range: [v1.43.1, v1.44.3)
- Port commit: 
-
https://github.com/MystenLabs/sui/commit/78499de65a5ab584ea4d568645a49f4fd1ad0005
- Description:

Add notify_read_synced_checkpoint/notify_read_executed_checkpoint

## Links to any relevant issues

Part of #5727 

## How the change has been tested

- [x] Basic tests (linting, compilation, formatting, unit/integration
tests)
- [ ] Patch-specific tests (correctness, functionality coverage)
- [ ] I have added tests that prove my fix is effective or that my
feature works
- [ ] I have checked that new and existing unit tests pass locally with
my changes

bingyanglin committed:
feat(node): Clean up various APIs (#8155)

# Description of change

- Upstream range: [v1.43.1, v1.44.3)
- Port commit: 
-
https://github.com/MystenLabs/sui/commit/437bfeee6d537926fa34e7760a780c2161f44a53
- Description:

Removes async and IotaResult from several APIs where it is not needed.

## Links to any relevant issues

Part of #5727 

## How the change has been tested

- [x] Basic tests (linting, compilation, formatting, unit/integration
tests)
- [ ] Patch-specific tests (correctness, functionality coverage)
- [ ] I have added tests that prove my fix is effective or that my
feature works
- [ ] I have checked that new and existing unit tests pass locally with
my changes

jkrvivian committed:
upstream(core): Remove redundant synced checkpoint notifications (#8163)

# Description of change

- Upstream range: [v1.43.1, v1.44.3)
- Port commit: 
-
https://github.com/MystenLabs/sui/commit/275464aafc15377ac9f129ccf34a56f6e4e9acaf
- Description:

Remove the redundant synced checkpoint notification from epoch store.

## Links to any relevant issues

Part of #5727 

## How the change has been tested

- [x] Basic tests (linting, compilation, formatting, unit/integration
tests)
- [ ] Patch-specific tests (correctness, functionality coverage)
- [ ] I have added tests that prove my fix is effective or that my
feature works
- [ ] I have checked that new and existing unit tests pass locally with
my changes

jkrvivian committed:
upstream(core): Rewrite and simplify CheckpointExecutor (#8170)

- Upstream range: [v1.43.1, v1.44.3)
- Port commit:
-
https://github.com/MystenLabs/sui/commit/fc0bad292ccf7bbb3e9308ecc37592602b56a7b6
- Description:

This simplifies CheckpointExecutor massively, in preparation for
optimizations which will be needed for throughput improvements

NOTE: We omitted the congestion tracker changes but updated
`crates/iota-core/src/checkpoints/checkpoint_executor/tests.rs`, since
we kept the test file while sui had removed it.

Part of #5727

- [x] Basic tests (linting, compilation, formatting, unit/integration
tests)
- [ ] Patch-specific tests (correctness, functionality coverage)
- [ ] I have added tests that prove my fix is effective or that my
feature works
- [ ] I have checked that new and existing unit tests pass locally with
my changes

jkrvivian committed:
upstream(core): Deprecate skip_to/skip_prior_to/skip_to_last/reverse methods (#8143)

# Description of change

- Upstream range: [v1.43.1, v1.44.3)
- Port commit: 
-
https://github.com/MystenLabs/sui/commit/0269b42498befb8f4cf628dd16b277d2e9d250dc



## Links to any relevant issues
Part of #5727 

## How the change has been tested

- [x] Basic tests (linting, compilation, formatting, unit/integration
tests)
- [ ] Patch-specific tests (correctness, functionality coverage)
- [ ] I have added tests that prove my fix is effective or that my
feature works
- [ ] I have checked that new and existing unit tests pass locally with
my changes

---------

Co-authored-by: muXxer <git@muxxer.de>

Thoralf-M commented on crates/iota-protocol-config/src/lib.rs:
```diff
@@ -1315,6 +1315,12 @@ impl ProtocolConfig {
     }
+
+    pub fn consensus_num_requested_prior_commits_at_startup(&self) -> u32 {
+        // TODO: this will eventually be the max of some number of other
+        // parameters.
```
This should stay like this for now?

jkrvivian replied:
Yes, we'll change this in the next range together with https://slipstreamer.iota.cafe/?commit=f8f7465560d84d6d46650d348de882afb2dae766 .

thibault-martinez replied:
Double checking, this function is already being used so you are definitely sure a 0 value is fine yeah?

jkrvivian replied:
Yes, it's fine, it'll work as the way it is now. And we will have a follow-up PR to update this value with the upstream commits:
* https://slipstreamer.iota.cafe/?commit=9f4304d8323c74b372fbe7405f9ee8786636b52b
* https://slipstreamer.iota.cafe/?commit=f8f7465560d84d6d46650d348de882afb2dae766

jkrvivian committed:
upstream(core): Fix expensive callsites of iterator (#8282)

# Description of change

- Upstream range: [v1.44.3, v1.45.3)
- Port commit: 
-
https://github.com/MystenLabs/sui/commit/b40bce37d61ea57859a02da816c3c240660df7ae
- Description:
This fixes expensive call sites caused by
https://github.com/MystenLabs/sui/commit/0269b42498befb8f4cf628dd16b277d2e9d250dc
([typed store] iterators: deprecate
skip_to/skip_prior_to/skip_to_last/reverse methods), #8143


## Links to any relevant issues

Part of #6153 

## How the change has been tested

- [x] Basic tests (linting, compilation, formatting, unit/integration
tests)
- [ ] Patch-specific tests (correctness, functionality coverage)
- [ ] I have added tests that prove my fix is effective or that my
feature works
- [ ] I have checked that new and existing unit tests pass locally with
my changes

---------

Co-authored-by: Vlad Semenov <vlad.semenov@iota.org>

jkrvivian committed:
upstream(core): Add consensus data quarantining (#8247)

# Description of change

- Upstream range: [v1.43.1, v1.44.3)
- Port commit: 
-
https://github.com/MystenLabs/sui/commit/e4db4a07c31bfcbcada149cc8f0ef6a73c261b13
- Description:
- All consensus output is held in memory until all checkpoints built
from that output have been committed. This ensures that the epoch db is
not corrupted in the event of a fork.
- Consensus therefore becomes the primary replay log for all crash
recovery.
- Additionally this means there are several tables that we no longer
need to write to, and their contents can live in memory.

## Links to any relevant issues

Part of #8234

## How the change has been tested

- [x] Basic tests (linting, compilation, formatting, unit/integration
tests)
- [ ] Patch-specific tests (correctness, functionality coverage)
- [ ] I have added tests that prove my fix is effective or that my
feature works
- [ ] I have checked that new and existing unit tests pass locally with
my changes

### Release Notes

- [ ] Protocol:
- [x] Nodes (Validators and Full nodes): - Add metrics
`consensus_quarantine_queue_size` and `shared_object_assignments_size`
- [ ] Indexer:
- [ ] JSON-RPC:
- [ ] GraphQL:
- [ ] CLI:
- [ ] Rust SDK:
- [ ] REST API:

jkrvivian committed:
upstream(core): Deprecate skip_to_last/reverse methods in starfish rocksdb_store.rs (#8393)

## Description
Apply upstream changes in https://github.com/iotaledger/iota/pull/8143
to starfish.

## Links to any relevant issues
Part of #5727 

## How the change has been tested

- [x] Basic tests (linting, compilation, formatting, unit/integration
tests)
- [ ] Patch-specific tests (correctness, functionality coverage)
- [ ] I have added tests that prove my fix is effective or that my
feature works
- [ ] I have checked that new and existing unit tests pass locally with
my changes
